From 4c1d347f94d8f758fd7a050c008e247de373a9db Mon Sep 17 00:00:00 2001
From: Arnd Bergmann <arnd@arndb.de>
Date: Wed, 30 Dec 2020 16:45:40 +0100
Subject: [PATCH 05/10] genirq: Export irq_check_status_bit

Changing some inline functions to use the new irq_check_status_bit
function out of line breaks calling them from loadable modules:

ERROR: modpost: "irq_check_status_bit" [drivers/perf/arm_spe_pmu.ko] undefined!

Export the function to make it work again. One can debate over
whether this should be EXPORT_SYMBOL() or EXPORT_SYMBOL_GPL(),
as it could be called from any module before and making it GPL-only
changes behavior. However all other symbols in this file are
EXPORT_SYMBOL_GPL(), so I went with that for consistency.

Fixes: fdd029630434 ("genirq: Move status flag checks to core")
Signed-off-by: Arnd Bergmann <arnd@arndb.de>
Link: https://lore.kernel.org/r/20201230154600.697832-1-arnd@kernel.org
Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
---
 kernel/irq/manage.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kernel/irq/manage.c b/kernel/irq/manage.c
index ab8567f32501..dec3f73e8db9 100644
--- a/kernel/irq/manage.c
+++ b/kernel/irq/manage.c
@@ -2859,3 +2859,4 @@ bool irq_check_status_bit(unsigned int irq, unsigned int bitmask)
 	rcu_read_unlock();
 	return res;
 }
+EXPORT_SYMBOL_GPL(irq_check_status_bit);
-- 
2.30.0

